stf/src/sgx.rs

```diff
@@ -259,16 +269,45 @@ impl Stf {
                     )
                         .encode(),
                 ));
+                increment_nonce(&account_incognito);
                 Ok(())
             }
             TrustedCall::balance_shield(who, value) => {
                 debug!("balance_shield({:x?}, {})", who.encode(), value);
-                Self::shield_funds(who, value)?;
+                if !validate_nonce(&who, call.nonce) {
+                    return Err(StfError::InvalidNonce(call.nonce))
+                }
+                Self::shield_funds(who.clone(), value)?;
+                increment_nonce(&who);
                 Ok(())
             }
+        }
+    )
+    }
+
+    pub fn account_nonce(ext: &mut State, account: &AccountId) -> Option<Vec<u8>> {
```

Comment:
this function should return `Index` and it should directly do the mapping to zero if `None`. Your usage in `rpc/api.rs`isn't elegant if you do it like this

Reply:
Agreed & fixed